### PR TITLE
User data migration for languagepacks

### PR DESCRIPTION
**Pull request by benjaoming**
Fix for #4003

Done and tested in 0.14.alpha7

### CODE REVIEW

**Comment by benjaoming** [comment]
The consequence of this is that languagepacks have to be re-downloaded, but assuming that:

1. It's harmless for people running 0.14 alphas
1. 0.13 users would re-download anyways

**Comment by benjaoming** [comment]
Interesting!

```
Traceback (most recent call last):
  File "/usr/share/kalite/python-packages/django/core/management/base.py", line 224, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/usr/share/kalite/python-packages/django/core/management/base.py", line 263, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python2.7/dist-packages/kalite/distributed/management/commands/initialize_kalite.py", line 56, in handle
    self.reinitialize_server()
  File "/usr/lib/python2.7/dist-packages/kalite/distributed/management/commands/initialize_kalite.py", line 48, in reinitialize_server
    initialize_content_caches()
  File "/usr/lib/python2.7/dist-packages/kalite/caching/__init__.py", line 78, in initialize_content_caches
    for lang in i18n.get_installed_language_packs(force=True).keys():
  File "/usr/lib/python2.7/dist-packages/kalite/i18n/__init__.py", line 314, in get_installed_language_packs
    INSTALLED_LANGUAGES_CACHE = _get_installed_language_packs()
  File "/usr/lib/python2.7/dist-packages/kalite/i18n/__init__.py", line 352, in _get_installed_language_packs
    logging.error("Error reading %s metadata (%s): %s" % (django_disk_code, metadata_filepath, e))
UnboundLocalError: local variable 'metadata_filepath' referenced before assignment
```

**Comment by benjaoming** [comment]
Also, depressing :)

```
            try:
                # Get the metadata
                metadata_filepath = os.path.join(locale_dir, django_disk_code, "%s_metadata.json" % lcode_to_ietf(django_disk_code))
                lang_meta = softload_json(metadata_filepath, raises=True)

                logging.debug("Found language pack %s" % (django_disk_code))
            except Exception as e:
                if isinstance(e, IOError) and e.errno == 2:
                    logging.info("Ignoring non-language pack %s in %s" % (django_disk_code, locale_dir))
                else:
                    logging.error("Error reading %s metadata (%s): %s" % (django_disk_code, metadata_filepath, e))
                continue
```

Some catch all exception block...

**Comment by benjaoming** [comment]
This was confirmed to work in alpha7, recommending a merge

**Comment by benjaoming** [comment]
Adding a merge of current 0.14.x branch to force a fresh CI test

**Comment by aronasorman** [comment]
Merge conflict.

File: kalite/legacy/i18n_settings.py [hunk 1]
```diff
@@ -54,10 +54,23 @@ def allow_all_languages_alist(langlookupfile):
...
 I18N_DATA_PATH = resource_filename("kalite", "i18n/data")
 LANG_LOOKUP_FILEPATH = os.path.join(I18N_DATA_PATH, "languagelookup.json")
 
+__dubbed_video_mapping_source = os.path.join(I18N_DATA_PATH, "dubbed_video_mappings.json")
+DUBBED_VIDEOS_MAPPING_FILEPATH = os.path.join(USER_WRITABLE_LOCALE_DIR, "dubbed_video_mappings.json")
+
+# If user does not have this file in their own directory, create a new one as a
+# copy of the distributed one since user processes may write to it
+if os.path.isfile(__dubbed_video_mapping_source) and not os.path.isfile(DUBBED_VIDEOS_MAPPING_FILEPATH):
+    open(DUBBED_VIDEOS_MAPPING_FILEPATH, "w").write(
```
Comment (aronasorman) [inline kalite/legacy/i18n_settings.py]: Couldn't you just copy the file with a function in `shutil`?

**Comment by MCGallaspy** [comment]
This is incorporated into #4093, right?

**Comment by benjaoming** [comment]
@MCGallaspy yes but the Alpha PRs aren't supposed to be merged, as that would short-circuit the individual processes of all the existing PRs that it includes.

**Comment by benjaoming** [comment]
no more conflicts @aronasorman

**Pull request merged**